sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/UnivocityParser.scala

```diff
@@ -202,21 +221,25 @@ private[csv] class UnivocityParser(
       }
       numMalformedRecords += 1
       None
-    } else if (options.failFast && schema.length != tokens.length) {
+    } else if (options.failFast && inputSchema.length != tokens.length) {
       throw new RuntimeException(s"Malformed line in FAILFAST mode: " +
         s"${tokens.mkString(options.delimiter.toString)}")
     } else {
-      val checkedTokens = if (options.permissive && schema.length > tokens.length) {
-        tokens ++ new Array[String](schema.length - tokens.length)
-      } else if (options.permissive && schema.length < tokens.length) {
-        tokens.take(schema.length)
+      val checkedTokens = if (options.permissive && inputSchema.length > tokens.length) {
+        tokens ++ new Array[String](inputSchema.length - tokens.length)
+      } else if (options.permissive && inputSchema.length < tokens.length) {
+        tokens.take(inputSchema.length)
```

Comment:
not related to this PR, but can you add some comments for this code block? It's kind if hard to follow the logic here